commitrmnocb/outcome.go

```diff
@@ -108,11 +123,23 @@ func buildReport(
 		outcomeType = ReportEmpty
 	}
 
+	var lastPricesUpdate time.Time
+	tokenPrices := consensusObservation.TokenPricesArray()
+
+	// If we're updating all token prices, set the last prices update to the observed timestamp,
+	// otherwise keep it as the last update time.
+	if len(offchainCfg.TokenInfo) == len(tokenPrices) {
+		lastPricesUpdate = consensusObservation.Timestamp
+	} else {
+		lastPricesUpdate = lastUpdate
```

Comment:
Is this field copied through in the outcomes of every state? Seems like in many cases `previousOutcome.LastPriceUpdate` could be nil.